### ISSUE TITLE
Upgrade to JUnit4 for testing Prolog in Java

### ISSUE THREAD

**Issue by ssardina** [SWI-Prolog/packages-jpl#40]
The current `org.jpl7.test.TestJUnit` is using old JUnit version 3.  That class is used to test Prolog inside Java.

Would be great to upgrade to JUnit version 4:

https://objectcomputing.com/resources/publications/sett/august-2007-migrating-from-junit-3-to-junit-4-nothing-but-good-news

https://junit.org/junit4/faq.html

Once that is done, `src/java/org/jpl7/test/TestOLD.java` could be integrated and removed and also integrate the standalone tests which are already done in JUnit4.

Note that the jar file for JUnit v4 is `junit4.jar`

**Comment by ssardina** [comment]
Java unit testing is now fully in JUnit v4. 

However, I will close this once I integrate 'src/java/org/jpl7/test/TestOLD.java' and all tests in dir `src/java/org/jpl7/test/standalone`

**Comment by JanWielemaker** [comment]
I totally agree.  Just create the PRs you think should be done.  Thanks!

**Comment by ssardina** [comment]
Ahhh OK I was going to rebase and merge into master, but I just created a PR #42 for you to review.

This was a detour of what I wanted to do with Rationals but I thought it was good to clean and update all this. I found a few other rough corners too that I will address over time (opened issues)

**Comment by JanWielemaker** [comment]
Ok.  If you think it is ready to be merged I'll check and merge.  Typically I do little for JPL except for checking no obviously strange things happen and the tests still pass.   I do frequently update the commit message to create a better ChangeLog entry and often merge commits that have little meaning in isolation.

Should I do so for this PR now?

**Comment by ssardina** [comment]
Yes, go ahead!

The tests are all passing on my side, but it's good you do a sanity check, go for it.

**Comment by ssardina** [comment]
There seems to be some issue with dependencies of the new JUNIT4

Check this: https://swi-prolog.discourse.group/t/java-jpl-cmake-test-errors/2177/3

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/java-jpl-cmake-test-errors/2177/4

**Comment by ssardina** [comment]
I am handling it there too and have extended CMAKE to be smarter at finding the JAR file. 

All should be ready for a new PR with a much cleaner and extensive testing suite, at last... :-)

**Comment by ssardina** [comment]
All issues here and [in forum](https://swi-prolog.discourse.group/t/java-jpl-cmake-test-errors/2177/21) fixed after commit 44bd4a0fc0963d6607e8fc1d4585dc2e72b13424
